python/pyspark/ml/torch/distributor.py

```diff
@@ -554,7 +561,21 @@ def set_gpus(context: "BarrierTaskContext") -> None:
                     pass
 
             if context.partitionId() == 0:
-                yield output
+                output_bytes = cloudpickle.dumps(output)
+                output_size = len(output_bytes)
+
+                # In Spark Connect, DataFrame.collect stacks rows to size
+                # 'spark.connect.grpc.arrow.maxBatchSize' (default 4MiB),
+                # here use 4KiB for each chunk, which mean each arrow batch
+                # may contain about 1000 chunks.
+                chunks = []
+                chunk_size = 4096
```

Comment:
I think we should use larger chunk size, I recommend 256MB .

Reply:
the arrow batch size is actually controlled by `spark.connect.grpc.arrow.maxBatchSize`, and if it exceeds the max size of grpc message it fails.
So I think we don't need a larger chunk size here

Reply:
Make sense.